probatus/utils/shap_helpers.py

```diff
@@ -85,31 +85,30 @@ def shap_calc(
         if verbose <= 100:
             warnings.simplefilter("ignore")
 
-        # Create the background data,required for non tree based models.
-        # A single datapoint can passed as mask (https://github.com/slundberg/shap/issues/955#issuecomment-569837201)
-
-        if X.shape[0] < sample_size:
-            sample_size = int(np.ceil(X.shape[0] * 0.2))
-        else:
-            pass
-        mask = sample(X, sample_size)
-
-        if X.select_dtypes("category").shape[1] > 0:
-            if verbose > 0:
-                warnings.warn(
-                    "Using tree_dependent feature_perturbation (in shap) without background"
-                    " data for tree-based model with categorical features."
-                )
+        # For tree explainers, do not pass masker when feature_perturbation is
+        # tree_path_dependent, or when X contains categorical features
+        # related to issue:
+        # https://github.com/slundberg/shap/issues/480
+        if shap_kwargs.get("feature_perturbation") == "tree_path_dependent" or X.select_dtypes("category").shape[1] > 0:
```

Comment:
Wasn't it that mask is only "required" for logistic regression @anilkumarpanda?

Reply:
Mask is required for non Tree-based models.
For Tree-based models you can have two types of `feature_perturbation`: `tree_path_dependent` and `interventional`. The former does not require a background dataset (aka mask), the latter does require it.
It is explained in the shap documentation:
https://shap.readthedocs.io/en/latest/generated/shap.explainers.Tree.html#shap.explainers.Tree.__init__
(see the `feature_perturbation` parameter)

Reply:
True,have a mask is required for non tree based models.